services/blockstorage/adapter/persistence.go

```diff
@@ -8,6 +8,7 @@ import (
 type BlockPersistence interface {
 	WriteBlock(blockPairs *protocol.BlockPairContainer) error
 	ReadAllBlocks() []*protocol.BlockPairContainer
+	ReadAllBlocksByTimeRange(start, end primitives.TimestampNano) []*protocol.BlockPairContainer
```

Comment:
if it's by time range, it's not all blocks. Maybe `FindBlocksWithinTimeRange`? also, this should _probably_ return a cursor so as to not allow your users to kill the system

Reply:
good point, fixing

Reply:
re: cursor
system should support pagination (i will add it to the spec and open a pr) but this specific request returns a specific result (like a boolean) and not a set of results
the search 'space', start->end is implementation based and not exposed by the external api, external api which currently calls it exposes a single timestamp

Reply:
nevertheless, someone contributing to the system or a new developer in the core team might think it's feasible (since the API supports it!) to retrieve a huge chunk of blocks this way.

Reply:
at the very least, please add a `maxNumOfResults` argument to the list, and document (godoc) that it should never exceed some `k` size.